src/components/Navbar/Navbar.jsx

```diff
@@ -0,0 +1,217 @@
+  }, boxRef);
+
+  return (
+    <>
+      <div className="navbar-mobile">
+        <Link to="#" className="menu-bars">
+          <List onClick={showSidebar} />
+        </Link>
+        <Link to="/" className="navbar-mobile-logo">
+          LOGO
+        </Link>
+      </div>
+      <nav ref={boxRef} className={sidebar ? "nav-menu active" : "nav-menu"}>
+        <ul className="nav-menu-items">
+          <li className="navbar-toggle" onClick={showSidebar}>
+            <Link to="#" className="menu-bars">
+              <List />
+            </Link>
+          </li>
+
+          <li className="nav-li nav-li-login-sign" onClick={showSidebar}>
+            <Link to="/logIn">
+              <Button
+                variant="primary"
+                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
+              >
+                Log In
+              </Button>
+            </Link>
+          </li>
+          <li className="nav-li nav-li-login-sign" onClick={showSidebar}>
+            <Link to="/signUp">
+              <Button
+                variant="primary"
+                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
+              >
+                Sign Up
+              </Button>
+            </Link>
+          </li>
+
+          <li className="nav-li" onClick={showSidebar}>
+            <Link to="/" className="nav-text">
+              <HouseDoor /> &nbsp; Home
+            </Link>
+          </li>
+          <li className="nav-li" onClick={showSidebar}>
+            <Link to={{ pathname: "/", hash: "#search" }} className="nav-text">
+              <Search /> &nbsp; Search
+            </Link>
+          </li>
+          <li className="nav-li" onClick={showSidebar}>
+            <Link to="/requests" className="nav-text">
+              <Gift /> &nbsp; Requests
+            </Link>
+          </li>
+          <li className="nav-li" onClick={showSidebar}>
+            <Link to="/donations" className="nav-text">
+              <Heart /> &nbsp; Donations
+            </Link>
+          </li>
+          <li className="nav-li" onClick={showSidebar}>
+            <Link to="/about" className="nav-text">
+              <InfoCircle /> &nbsp; About
+            </Link>
+          </li>
+          <li className="nav-li" onClick={showSidebar}>
+            <Link to="/contactUs" className="nav-text">
+              <Envelope /> &nbsp; Contact Us
+            </Link>
+          </li>
+          <li className="nav-li nav-li-lang">
+            <GeoAlt style={{ color: "black" }} />
+            <DropdownButton id="dropdown-basic-button" title="Language">
+              <Dropdown.Item href="#/action-1" onClick={showSidebar}>
+                TUR
+              </Dropdown.Item>
+              <Dropdown.Item href="#/action-2" onClick={showSidebar}>
+                ENG
+              </Dropdown.Item>
+            </DropdownButton>
+          </li>
+        </ul>
+      </nav>
+    </>
+  );
+};
+
+const getWindowDimensions = () => {
```

Comment:
this function and useWindowDimension function can be moved into a general folder called hooks so it can be reused with all the other components that needs to use them